Jetpack Cloud: Show the correct icon in the "Add your server credentials" banner

mattgawarecki opened the pull request:
#### Changes proposed in this Pull Request

* Add the `iconPath` property to the `Banner` component, so that it can accept non-Gridicon images as the banner icon.
* Move the "connected" and "disconnected" Jetpack icons into the shared assets directory, so that they can be shared among multiple components.
* Set the `iconPath` in `JetpackBackupCredsBanner` to display the "disconnected" icon instead of the "history" Gridicon.

#### Testing instructions

##### On WordPress.com

* Select a Jetpack site with Backup that does not have stored credentials, or has stored invalid credentials.
* Verify that the "Add your server credentials" banner appears at the top of the **Activity** page with the same icon as shown in the screenshots below. Notably, it should *not* be the "history" Gridicon.

##### On Jetpack.com

* Verify that the "connected" and "disconnected" icons still display properly on the **Settings** page (see screenshots).

#### Screenshots

##### WordPress.com

<img width="784" alt="Screen Shot 2020-07-28 at 13 26 12" src="https://user-images.githubusercontent.com/670067/88706168-19a45400-d0d6-11ea-938c-8a1198c3cb6d.png">

<img width="391" alt="Screen Shot 2020-07-28 at 13 25 55" src="https://user-images.githubusercontent.com/670067/88706185-1f9a3500-d0d6-11ea-9386-df02fda02b9e.png">

##### Jetpack.com

<img width="391" alt="Screen Shot 2020-07-28 at 13 28 57" src="https://user-images.githubusercontent.com/670067/88706350-5ec88600-d0d6-11ea-9193-046b060b8663.png">

<img width="391" alt="Screen Shot 2020-07-28 at 13 33 37" src="https://user-images.githubusercontent.com/670067/88706788-f5954280-d0d6-11ea-87db-49483af570ab.png">

rcanepa commented on client/blocks/jetpack-backup-creds-banner/style.scss:
```diff
@@ -0,0 +1,3 @@
+.jetpack-backup-creds-banner.card.banner {
+	@include banner-color( #D63638 );
```
I think we can use `var( --studio-red-50 )` instead of `#D63638`.

mattgawarecki commented:
Added Renzo's color suggestion (thanks for that!) and adjusted the size of the banner icon:

<img width="388" alt="Screen Shot 2020-07-29 at 14 55 21" src="https://user-images.githubusercontent.com/670067/88846775-b258d380-d1ab-11ea-8b51-1db4d9772608.png">

<img width="847" alt="Screen Shot 2020-07-29 at 14 55 37" src="https://user-images.githubusercontent.com/670067/88846789-b5ec5a80-d1ab-11ea-87cb-afd8dcc4d80f.png">
